pytest_html/extras.py

```diff
@@ -9,16 +9,32 @@
 FORMAT_URL = 'url'
 
 
-def extra(content, format, name=None):
-    return {'name': name, 'format': format, 'content': content}
+def extra(content, format, name=None, mime_type='image/png', extension='png'):
```

Comment:
Here the `mime_type` and `extension` should default to `None`. Then we can set the appropriate values in `image`, `png`, `jpg`, `svg`, `json`, and `text`.